src/components/CodeSnippet/CodeSnippet.tsx

```diff
@@ -0,0 +1,143 @@
+import classNames from "classnames";
+import PropTypes from "prop-types";
+import React from "react";
+import type { ReactNode } from "react";
+
+export type DropdownOptionPros = {
+  label: string;
+  value: string | number;
+  selected?: boolean;
+  disabled?: boolean;
+};
+
+export type CodeSnippetDropdownProps = {
+  onChange?: (evt: React.SyntheticEvent) => void;
+  options: Array<DropdownOptionPros>;
+};
+
+export const CodeSnippetDropdown = ({
+  options,
+  onChange,
+}: CodeSnippetDropdownProps): JSX.Element => {
+  return (
+    <select className="p-code-snippet__dropdown" onChange={onChange}>
+      {options.map(({ label, value, ...props }) => (
+        <option value={value} key={value || label} {...props}>
+          {label}
+        </option>
+      ))}
+    </select>
+  );
+};
+
+CodeSnippetDropdown.props = {
+  onChange: PropTypes.func,
+  options: PropTypes.arrayOf(
+    PropTypes.shape({
+      label: PropTypes.string.isRequired,
+      value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
+    })
+  ).isRequired,
+};
+
+export const CodeSnippetBlock = ({
+  code,
+  title,
+  numbered = false,
+  icon = null,
+  isWrapped = false,
+  dropdowns,
+}: CodeSnippetBlockProps): JSX.Element => {
+  let className = "p-code-snippet__block";
+  let numberedCode;
+
+  if (numbered) {
+    className += "--numbered";
+
+    // wrap code lines in spans (and preserve the whitespace)
+    const lines = code.split(/\r?\n/);
+    numberedCode = lines.map((line, i) => (
+      <React.Fragment key={`p-code-snippet__line-${i}`}>
+        <span className="p-code-snippet__line">{line}</span>
+        {"\n"}
+      </React.Fragment>
+    ));
+  } else if (icon) {
```

Comment:
Do we really want to special case `icon` like this? Won't this preclude this use case...
```
$ cd foo
$ ./run something
```

A more generic approach might be beneficial here. Where you run any code provided through the spanification above and add the appropriate icon to the start of the line.

Reply:
In terms of adding an icon to a code block it was designed to have one icon, not an icon per block.
So technically (at least in a way it was implemented in Vanilla SCSS) it is a separate "special case" modifier on a block.